### PR TITLE
Update the NodeJS version compat checks.

### PR DESCRIPTION
**Pull request by pgavlin**
- Unify the 1.x.y series and the 0.17.z series
- Fix the check s.t. post-1.0, only the major versions are required to
  match

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+## 1.0.0-beta.2 (2019-08-13)
```
Comment (pgavlin) [inline CHANGELOG.md]: I will tag 1.0.0-beta.2 as soon as this is in.

File: sdk/nodejs/cmd/pulumi-language-nodejs/main.go [hunk 2]
```diff
@@ -160,6 +160,31 @@ func newLanguageHost(nodePath, runPath, engineAddress,
...
+			return false, "Differing major or minor versions are not supported."
+		}
+
+	case a.Major >= 1 && b.Major >= 1:
+		// If both major versions are post-1.0, we require that the major versions match.
+		if a.Major != b.Minor {
+			return false, "Differing major versions are not supported."
+		}
+
+	case a.Major == 1 && b.Minor == 17 || b.Major == 1 && a.Minor == 17:
```
Comment (ellismg) [inline sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: Might want to confirm that when you check the minor for 17, the major is 0 here, right?

Comment (lukehoban) [inline sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: Technically it feels like this needs an `&& b.Major == 0`, though that will only matter when we get to `2.17`?

**Comment by pgavlin** [comment]
I have manually verified that this works as expected with a mix of 1.x.y and 0.17.z series packages.

File: sdk/nodejs/cmd/pulumi-language-nodejs/main.go [hunk 3]
```diff
@@ -160,6 +160,31 @@ func newLanguageHost(nodePath, runPath, engineAddress,
...
+	switch {
+	case a.Major == 0 && b.Major == 0:
+		// If both major versions are pre-1.0, we require that the major and minor versions match.
+		if a.Minor != b.Minor {
+			return false, "Differing major or minor versions are not supported."
+		}
+
+	case a.Major >= 1 && b.Major >= 1:
+		// If both major versions are post-1.0, we require that the major versions match.
+		if a.Major != b.Minor {
```
Comment (lukehoban) [inline sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: `b.Major`?

Comment (pgavlin) [reply sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: Ah, good catch

File: sdk/nodejs/cmd/pulumi-language-nodejs/main.go (hunk 2 above)
Comment (pgavlin) [reply sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: It doesn't: if we get to this case, we know that one version has a major version of zero and the other has a major version greater than zero.

**Comment by lukehoban** [comment]
> The goal of these checks where you ensure that you didn't end up with multiple copies of @pulumi/pulumi in your node_modules folder

Strictly speaking that wasn't quite what this code guaranteed - it just verified that you can't have sxs "across different minor versions".  This was because we assumed that when we make breaking changes, we will likely not be able to support SxS correctly.  

The changes here maintain this - they just:
1. Treat `0.17.x` as equal to `1.0.0` for purposes of this check, and
2. Change the "different minor" to "different major" for post-1.0 in line with the general change that "breaking" is major bump after 1.0 but was minor bump before 1.0

File: sdk/nodejs/cmd/pulumi-language-nodejs/main.go (hunk 2 above)
Comment (pgavlin) [reply sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: No need, but I've added a check for clarity.

Comment (pgavlin) [reply sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: Added a clause for clarity.

**Pull request merged**